Unable to initialize main class com.intel.analytics.zoo.serving.http.FrontEndApp

在graphe下面跑下述的java命令的时候，有报错提示：
[报错提示]
Error: Unable to initialize main class com.intel.analytics.zoo.serving.http.FrontEndApp
Caused by: java.lang.NoClassDefFoundError: scala/collection/Seq 

[Java命令]java \
    -Xms1g \
    -Xmx2g \
    -Dcom.intel.analytics.zoo.shaded.io.netty.tryReflectionSetAccessible=true \
    -Dakka.actor.default-dispatcher.fork-join-executor.parallelism-factor=2 \
    -jar ./scripts/analytics-zoo/zoo/target/analytics-zoo-bigdl_0.11.1-spark_2.4.3-0.9.0-SNAPSHOT-http.jar \
    --redisHost '172.168.10.103' \
    --httpsEnabled true \
    --httpsKeyStorePath '/home/sdp/test5/narwhal/Trusted_ML/Trusted_Cluster_Serving/bin/keys/keystore.pkcs12' \
    --httpsKeyStorePassword '1234qwer' \
    --redisSecureEnabled false \
    --redissTrustStorePath '/home/sdp/test5/narwhal/Trusted_ML/Trusted_Cluster_Serving/bin/keys/keystore.jks' \
    --redissTrustStorePassword '1234qwer'

似乎与az的jar包和scala的path有关

--- a/zoo/pom.xml
+++ b/zoo/pom.xml
@@ -20,7 +20,7 @@
         <spark-scope>provided</spark-scope>
         <bigdl-scope>compile</bigdl-scope>
 
-        <scala-library-scope>provided</scala-library-scope>
+        <scala-library-scope>compile</scala-library-scope>
         <scala.major.version>2.11</scala.major.version>
         <scala.version>2.11.8</scala.version>
         <scala.macros.version>2.1.0</scala.macros.version>
@@ -153,7 +153,6 @@
             <id>all-in-one</id>
             <properties>
                 <spark-scope>compile</spark-scope>
-                <scala-library-scope>compile</scala-library-scope>
             </properties>
         </profile>
 
@@ -707,6 +706,11 @@
                         <configuration>
                             <shadedArtifactAttached>true</shadedArtifactAttached>
                             <shadedClassifierName>jar-with-dependencies</shadedClassifierName>
+                            <artifactSet>
+                                <excludes>
+                                    <exclude>org.scala-lang:scala-library</exclude>
+                                </excludes>
+                            </artifactSet>
                         </configuration>
                     </execution>
 
@@ -732,27 +736,13 @@
                                 </transformer>
                             </transformers>
                             <artifactSet>
-                                <excludes>
-                                    <exclude>com.intel.analytics.bigdl:bigdl-SPARK_2.4</exclude>
-                                    <exclude>ml.dmlc:xgboost4j</exclude>
-                                    <exclude>org.spark-project.spark:unused</exclude>
-                                    <exclude>org.tensorflow:libtensorflow</exclude>
-                                    <exclude>org.tensorflow:proto</exclude>
-                                    <exclude>black.ninia:jep</exclude>
-                                    <exclude>org.tensorflow:tensorflow-hadoop</exclude>
-                                    <exclude>com.intel.analytics.zoo:zoo-core-dist-all</exclude>
-                                    <exclude>com.intel.analytics.zoo:zoo-core-openvino-java-linux</exclude>
-                                    <exclude>com.intel.analytics.zoo:zoo-core-pmem-java-linux</exclude>
-                                    <exclude>com.intel.analytics.zoo:zoo-core-tfnet-linux</exclude>
-                                    <exclude>com.intel.analytics.zoo:zoo-core-mkl-linux</exclude>
-                                    <exclude>com.intel.analytics.zoo:zoo-core-dist-mac</exclude>
-                                    <exclude>com.intel.analytics.zoo:zoo-core-tfnet-mac</exclude>
-                                    <exclude>com.intel.analytics.zoo:zoo-core-openvino-java-mac</exclude>
-                                    <exclude>com.intel.analytics.bigdl.core.dist:all</exclude>
-                                    <exclude>com.intel.analytics.bigdl.spark-version</exclude>
-                                    <exclude>org.scalanlp:breeze_2.11</exclude>
-                                    <exclude>it.unimi.dsi:fastutil</exclude>
-                                </excludes>
+                                <includes>
+                                    <include>com.typesafe.akka</include>
+                                    <include>io.dropwizard.metrices</include>
+                                    <include>org.slf4j</include>
+                                    <include>com.google.guava:guava</include>
+                                    <include>org.scala-lang:scala-library</include>
+                                </includes>
                             </artifactSet>
                         </configuration>
                     </execution>

--- a/zoo/pom.xml
+++ b/zoo/pom.xml
@@ -738,10 +738,16 @@
                             <artifactSet>
                                 <includes>
                                     <include>com.typesafe.akka</include>
-                                    <include>io.dropwizard.metrices</include>
+                                    <include>io.dropwizard.metrics:metrecs-core</include>
                                     <include>org.slf4j</include>
+                                    <include>log4j:log4j</include>
                                     <include>com.google.guava:guava</include>
+                                    <include>com.google.flatbuffers:flatbuffers-java</include>
                                     <include>org.scala-lang:scala-library</include>
+                                    <include>com.github.scopt</include>
+                                    <include>com.typesafe:config</include>
+                                    <include>com.fasterxml.jackson.module</include>
+                                    <include>com.fasterxml.jackson.core</include>
                                 </includes>
                             </artifactSet>
                         </configuration>

--- a/zoo/pom.xml
+++ b/zoo/pom.xml
@@ -736,19 +736,27 @@
                                 </transformer>
                             </transformers>
                             <artifactSet>
-                                <includes>
-                                    <include>com.typesafe.akka</include>
-                                    <include>io.dropwizard.metrics:metrecs-core</include>
-                                    <include>org.slf4j</include>
-                                    <include>log4j:log4j</include>
-                                    <include>com.google.guava:guava</include>
-                                    <include>com.google.flatbuffers:flatbuffers-java</include>
-                                    <include>org.scala-lang:scala-library</include>
-                                    <include>com.github.scopt</include>
-                                    <include>com.typesafe:config</include>
-                                    <include>com.fasterxml.jackson.module</include>
-                                    <include>com.fasterxml.jackson.core</include>
-                                </includes>
+                                <excludes>
+                                    <exclude>com.intel.analytics.bigdl:bigdl-SPARK_2.4</exclude>
+                                    <exclude>ml.dmlc:xgboost4j</exclude>
+                                    <exclude>org.spark-project.spark:unused</exclude>
+                                    <exclude>org.tensorflow:libtensorflow</exclude>
+                                    <exclude>org.tensorflow:proto</exclude>
+                                    <exclude>black.ninia:jep</exclude>
+                                    <exclude>org.tensorflow:tensorflow-hadoop</exclude>
+                                    <exclude>com.intel.analytics.zoo:zoo-core-dist-all</exclude>
+                                    <exclude>com.intel.analytics.zoo:zoo-core-openvino-java-linux</exclude>
+                                    <exclude>com.intel.analytics.zoo:zoo-core-pmem-java-linux</exclude>
+                                    <exclude>com.intel.analytics.zoo:zoo-core-tfnet-linux</exclude>
+                                    <exclude>com.intel.analytics.zoo:zoo-core-mkl-linux</exclude>
+                                    <exclude>com.intel.analytics.zoo:zoo-core-dist-mac</exclude>
+                                    <exclude>com.intel.analytics.zoo:zoo-core-tfnet-mac</exclude>
+                                    <exclude>com.intel.analytics.zoo:zoo-core-openvino-java-mac</exclude>
+                                    <exclude>com.intel.analytics.bigdl.core.dist:all</exclude>
+                                    <exclude>com.intel.analytics.bigdl.spark-version</exclude>
+                                    <exclude>org.scalanlp:breeze_${scala.major.version}</exclude>
+                                    <exclude>it.unimi.dsi:fastutil</exclude>
+                                </excludes>
                             </artifactSet>
                         </configuration>
                     </execution>

